pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotRealtimeSegmentManager.java

```diff
@@ -201,15 +201,13 @@ private synchronized void assignRealtimeSegmentsToServerInstancesIfNecessary() {
           for (String partition : state.getPartitionSet()) {
             // Helix partition is the segment name
             if (SegmentName.isHighLevelConsumerSegmentName(partition)) {
-              HLCSegmentName segName = new HLCSegmentName(partition);
-              RealtimeSegmentZKMetadata realtimeSegmentZKMetadata = ZKMetadataProvider
-                  .getRealtimeSegmentZKMetadata(_pinotHelixResourceManager.getPropertyStore(), segName.getTableName(),
-                      partition);
-              if (realtimeSegmentZKMetadata == null) {
+              SegmentZKMetadata segmentZKMetadata = ZKMetadataProvider
```

Comment:
looks like segName was not used in the old code. but should it be used actually? segName vs. partition in the getSegmentZKMetadata() call

Reply:
partition is the same as segment name (mentioned in the comment)